Solving for the amount to double in savings based on compound interest. 

Compound interest means that the interest rate gets divided by the kind of compound throughout a year. For example, if it's compounded annually, that means that we divide the rate of interest by 12 because we want the rate to be the same every single month. 

So to calculate how long it would take for a certain principle amount to double we use:

$$ A_{amount} = P_{principle}(1 + \frac{r}{n})^{nt} $$ 

In particular we want:

$$ 2P = P(1 + \frac{0.05}{12})^{12t} $$ 

$$ \frac{2P}{P} = (1 + \frac{0.05}{12})^{12t} $$ 

$$ 2 = (1 + \frac{0.05}{12})^{12t} $$ 

$$ 12t * log(1 + \frac{0.05}{12}) = log(2) $$ 

$$ 12t = \frac{log(2)}{log(1 + \frac{0.05}{12})} $$ 

In [2]:
import math 

print(math.log(2))
print(math.log(1 + (0.05/12)))


0.6931471805599453
0.004158010148663677


$$ log(2) \approx 0.693 $$
$$ log(1 + \frac{0.05}{12}) \approx 0.004 $$ 

$$ 12t = \frac{0.693}{0.004} $$ 

In [3]:
0.693 / 0.004

173.24999999999997

$$ t \approx \frac{173.245}{12} $$ 

In [4]:
173.245 / 12

14.437083333333334

$$ t \approx 14.437 $$ 

At what interest rate (to the nearest hundredth of a percent) compounded annually will money in savings double in five years? 

$$ 2 = (1 + \frac{r}{12})^{12t} $$

$$ t = 5 $$ 

$$ 2 = (1 + \frac{r}{12})^{12*5 \rightarrow 60} $$

$$ \sqrt[60]{2} = \sqrt[60]{((1 + \frac{r}{12}))^{60}} $$ 



In [6]:
left = math.pow(2,(1/60))
print(left)

1.0116194403019225


$$ \sqrt[60]{2} \approx 1.011619 $$ 

$$ 1.011619 = (1 + \frac{r}{12}) $$ 

$$ 1.011619 - 1 = \frac{r}{12} $$ 

$$ r = 12 * (1.011619 - 1) $$ 



In [7]:
12 * (1.011619 - 1)

0.13942800000000055

$$ r \approx 0.14 $$ 

$$ 3000 = P(1 + 0.04)^3 $$ 

$$ P = \frac{3000}{(1 + 0.04)^3} $$ 

In [8]:
P = 3000 / math.pow(1 + 0.04, 3)
print(P)

2666.9890760127446


$$ P \approx $ 2667 $$ 

In [9]:
2000*math.pow((1 + 0.06), 19)

6051.199004175184

In [10]:
2000*math.pow((1 + 0.065), 19)

6617.173828202582

In [11]:
2 * 6051 


12102

$$ 12102 = 2000(1.06)^{t} $$ 

In [12]:
12102 / 2000

6.051

In [14]:
math.log(6.051)/math.log(1.06)

30.895096640732298

In [15]:
2 * 6617

13234

In [16]:
13234 / 2000

6.617

In [17]:
math.log(6.617)/math.log(1.065) 

30.00632189691071

$$ V = C(1 -r)^{t} $$ 

$$ $12,000 = $25,000(1 - r)^{4.5} $$ 

In [18]:
12000/25000

0.48

In [19]:
math.pow(0.48,(1/4.5))

0.8495026136956846

$$ \frac{12000}{25000} = (1-r)^{4.5} $$
$$ 0.48 = (1-r)^{4.5} $$ 
$$ r = 1 - \sqrt[4.5]{0.48} $$ 

$$ r \approx 15  $$ 

$$ A(t) = A_{0}e^{-kt} $$ 

$$ 0.5 = e^{-k(5730)} $$ 

$$ k = \frac{ln(0.5)}{-5730} $$ 

In [20]:
math.log(0.5) / -5730 

0.00012096809433855938

In [23]:
math.log(160) / math.log(80)

1.1581795909397823

In [24]:
log(1000)

NameError: name 'log' is not defined

In [25]:
math.log(1000)

6.907755278982137